httpserver/http_connection.cpp

```diff
@@ -601,40 +632,45 @@ void connection_t::poll_db(std::string pk, std::string last_hash) {
     const bool lp_requested =
         request_.find("X-Loki-Long-Poll") != request_.end();
 
-    if (items.empty() && lp_requested && long_polling_counter < DB_POLL_LIMIT) {
+    if (!items.empty()) {
+        BOOST_LOG_TRIVIAL(trace)
+            << "Successfully retrieved messages for " << pk.substr(0, 2)
+            << "..." << pk.substr(pk.length() - 3, pk.length() - 1);
+    }
 
-        // initiate a timer to poll for new messages
-        long_polling_counter += 1;
-        poll_timer_.expires_after(DB_POLL_INTERVAL);
-        poll_timer_.async_wait(std::bind(&connection_t::poll_db,
-                                         shared_from_this(), pk, last_hash));
+    if (items.empty() && lp_requested) {
 
-    } else {
+        auto self = shared_from_this();
 
-        json res_body;
-        json messages = json::array();
+        // TODO: make sure we don't register the same pk twice;
```

Comment:
I don't think this comment is quite right because we discussed the other day that for multiple devices you would want to be able to register the same pubkey twice
Not sure if it is possible but what we really want is to make sure we don't register more than one connection from the same source?

Reply:
Maybe a soft solution could be just an arbitrary cap on the number of connections per pubkey?
Like 5 or something?

Reply:
Hm, not sure what the comment was about, so I will remove it. I don't like setting the limit on pubkeys since you'd be able to abuse it and deny the service for the actual owner of the pubkey.

Reply:
Ah of course, free reign it is then

Reply:
removed